poucave/utils.py

```diff
@@ -363,21 +365,19 @@ def on(self, event, callback):
         self.callbacks.setdefault(event, []).append(callback)
 
 
-_bqclient = None
-
-
 async def fetch_bigquery(sql):  # pragma: nocover
     """
     Execute specified SQL and return rows.
     """
+    bqclient = getattr(threadlocal, "bqclient", None)
```

Comment:
This retrieves the client on the thread spawning the job, rather than the thread running the job. I'd think that code needs to be moved inside the job function.